android/cli/hooks/aar-transform.js

```diff
@@ -481,3 +446,96 @@ class SimpleFileCache {
 		fs.writeFileSync(this.cachePathAndFilename, dataToWrite);
 	}
 }
+
+/**
+ * Hyperloop versions below 2.2.0 have their own AAR handling which we need to
+ * disable by hooking into the affected hooks and reverting the changes Hyperloop
+ * made.
+ *
+ * @param {Object} cli CLI instance
+ * @param {Object} builder Builder instance
+ * @param {Object} logger Logger instance
+ */
+function registerHyperloopCompatibilityFixes(cli, builder, logger) {
+	var hyperloopModule = null;
+	builder.nativeLibModules.forEach(function (module) {
+		if (module.id === 'hyperloop' && appc.version.lt(module.version, '2.2.0')) {
+			hyperloopModule = module;
+		}
+	});
+	if (hyperloopModule === null) {
+		return;
+	}
+
+	var hyperloopBuildPath = path.join(builder.projectDir, 'build/hyperloop/android');
+
+	cli.on('build.android.aapt', {
+		priority: 1100,
+		/**
+		 * Remove parameters passed to AAPT which are not required anymore since 6.1.0
+		 *
+		 * @param {Object} data Hook data
+		 * @param {Function} callback Callback function
+		 */
+		pre: function(data, callback) {
+			logger.trace('Cleaning AAPT options from changes made by Hyperloop');
+			var aaptOptions = data.args[1];
+			var extraPackagesIndex = aaptOptions.indexOf('--extra-packages') + 1;
```

Comment:
This is insufficient. `indexof()` returns `-1` if `--extra-packages` is not found. You have no error handling around this. If it is not found, then `extraPackages` will always be set to `aaptOptions[0]`.